x-pack/plugins/cases/public/containers/utils.ts

```diff
@@ -30,20 +30,80 @@ import {
   CaseUserActionsResponseRt,
   CommentType,
   CasePatchRequest,
+  CaseUserActionConnectorRt,
+  CaseConnector,
+  ConnectorTypes,
+  noneConnectorId,
+  CaseFullExternalService,
+  CaseUserActionExternalServiceRt,
 } from '../../common';
 import { AllCases, Case, UpdateByKey } from './types';
 import * as i18n from './translations';
 
 export const getTypedPayload = <T>(a: unknown): T => a as T;
 
-export const parseString = (params: string) => {
+export const parseStringAsConnector = (
+  id: string | null,
+  encodedData: string | null
+): CaseConnector | undefined => {
+  if (encodedData == null) {
+    return;
+  }
+
+  const decodedConnector = parseString(encodedData);
+
+  if (!CaseUserActionConnectorRt.is(decodedConnector)) {
+    return;
+  }
+
+  if (id == null && decodedConnector.type === ConnectorTypes.none) {
```

Comment:
When the connector is updated (or created with the default none connector) to the none connector the `id` field will be `null`. This is because when the backend is looking at the references it will not find a reference for the none connector. So the backend will mark the connector id to `null`. We can't mark it as `none` in the backend without decoding the user action's value which has performance concerns.

This function which is executed in the UI will determine if it is a none connector by looking at the type and populating the `id` appropriately.